code/game/objects/objs.dm

```diff
@@ -208,22 +208,17 @@
 
 /obj/set_dir(ndir)
 	. = ..()
-	if(directional_offset)
-		update_directional_offset()
-
-/obj/Move()
-	. = ..()
-	if(directional_offset)
+	if(directional_offset && anchored)
```

Comment:
I'm with Loaf on this: what actually moves/forceMoves anchored entities, except e.g. shuttle move and qdel?

Reply:
Got rid of the stuff in move a while ago. Also I don't think you're commenting on the right line?

Reply:
Well, it's reasonable to point this comment at any of the three. I just don't buy that this event is in the right place (anchored + forceMove/dir set). Things being moved or rotated while anchored are inherently ad hoc/outside the world model and will vary obj by obj, so it's not clear what you are intercepting here. I get that it probably works for this obj in particular, but this is an /obj-level forceMove override, it needs to be well thought out. I think Loaf's suggestion to have some ability to activate/deactivate offsets makes a lot more sense.

Reply:
The comment on the check being in move() made sense, since an anchored object never moves anyways, so I fixed that. ForceMove on the other hand doesn't work the same way, and will move the thing whether its anchored or not, that's how things are moved into the inventory for one. 
Also, you already can activate/deactivate the offset? Just null out the directional_offset var?

Then, the only objects using this are wall-mounted objects and items. I could check all of them if they want to have a directional offset, by calling a proc in the if condition, but that's getting a bit overkill when the only current, and unlikely to change, use case is wall-mounted objects.. I just don't see how that would be worth it, or not annoy people because of an extra unavoidable proc call inside a busy code path?

Reply:
If you can activate/deactivate it, then why not do it here as appropriate? That would be natural, and there are anchored setters that you can put the activation into.

On the one hand, you are arguing that `anchored` is the right condition for these wall objs and they are the only things using the system, so it's the right condition overall. On the other hand, you are writing the system at `/movable` level. I just don't think you can have it both ways.

Reply:
Changed the code since then.